native/src/modules/endpoint/sagas/fetchResourceCache.js

```diff
@@ -19,6 +19,26 @@ const createErrorMessage = (fetchResult: FetchResultType) => {
     `${message}'Failed to download ${result.url} to ${path}': ${result.errorMessage}\n`, '')
 }
 
+function * watchOnProgress (): Saga<void> {
+  const channel = new FetcherModule().createProgressChannel()
+  try {
+    let progress = 0
+    while (progress < 1) {
```

Comment:
Shouldn't we dispatch an action with `progress: 1` as well?

Reply:
Exactly, if the content is loaded an event to show the page is already dispatched, so if event if there is 100% there is little time to see it before the loading display is hidden.
I wanted to be sure that this event really stops BEFORE the "Loading Finished" event is dispatched.
When the Loading Finished event is dispatched progress will be set to 1 anyways

Perhaps it would be nicer in the way we already discussed, an own Type "loading" beside of "ready" and "error".
The cache "loading" type would contain "progress: number",  "ready" would only contain the loaded value (like before), not the progress value

Reply:
... but I suggest this is "future music" like we Germans say ;)